site/app/libraries/socket/Server.php

```diff
@@ -213,7 +213,7 @@ public function onMessage(ConnectionInterface $from, $msgString) {
 
         $msg = json_decode($msgString, true);
 
-        if ($msg["type"] === "new_connection") {
+        if ($msg["type"] === "new_connection" && isset($msg['page'])) {
             if (!array_key_exists($msg['page'], $this->clients)) {
```

Comment:
If `$msg['page']` is not set, we will end up in the `else` block and call `$this->broadcast()`.  Is that desirable in this case?

Reply:
No it's not, as broadcast needs the page the client is on which is **not set**